examples/gno.land/r/gov/dao/v3/impl/prop_requests.gno

```diff
@@ -0,0 +1,216 @@
+	if member == nil {
+		panic("proposer is not a member")
+	}
+
+	if member.InvitationPoints <= 0 {
+		panic("proposer does not have enough invitation points for inviting new people to the board")
+	}
+
+	return proxy.ProposalRequest{
+		Title:       "New Member porposal",
+		Description: "This prpoposal is looking to add a new member to the board.",
+		Callback:    addMemberCallback,
+		Metadata: &AddMemberMetadata{
+			Member:    memberByTier(tier),
+			Portfolio: portfolio,
+			Tier:      tier,
+			Addr:      addr,
+			Caller:    member,
+		},
+	}
+}
+
+type RemoveMemberMetadata struct {
+	Addr   std.Address
+	Tier   string
+	Reason string
+}
+
+func (a *RemoveMemberMetadata) IsMetadata() {}
+func (a *RemoveMemberMetadata) String() string {
+	return ufmt.Sprintf("Member with address %v will be withdrawn.\n\n REASON: %v.", a.Addr, a.Reason)
+}
+
+func removeMemberCallback(m proxy.Metadata) error {
+	rmm, ok := m.(*RemoveMemberMetadata)
+	if !ok {
+		panic("metadata type for removing a member is wrong")
+	}
+
+	memberstore.Get().RemoveMember(rmm.Addr)
+
+	return nil
+}
+
+func NewWithdrawMemberRequest(addr std.Address, reason string) proxy.ProposalRequest {
+	member, tier := memberstore.Get().GetMember(addr)
+	if member == nil {
+		panic("user we want to remove not found")
+	}
+
+	if tier == memberstore.T1 && reason == "" {
+		panic("T1 user removals must contains a reason.")
+	}
+
+	return proxy.ProposalRequest{
+		Title:       "Member Withdrawn",
+		Description: "This prpoposal is looking to remove a member from the board.",
+		Callback:    removeMemberCallback,
+		Metadata: &RemoveMemberMetadata{
+			Addr:   addr,
+			Tier:   tier,
+			Reason: reason,
+		},
+	}
+}
+
+type PromoteMemberMetadata struct {
+	FromTier string
+	ToTier   string
+	Addr     std.Address
+	Member   *memberstore.Member
+}
+
+func (a *PromoteMemberMetadata) IsMetadata() {}
+func (a *PromoteMemberMetadata) String() string {
+	return ufmt.Sprintf("A new member with address %v will be promoted from tier %v to tier %v.", a.Addr, a.FromTier, a.ToTier)
+}
+
+func promoteMemberCallback(m proxy.Metadata) error {
+	pmm, ok := m.(*PromoteMemberMetadata)
+	if !ok {
+		panic("metadata type for promoting a member is wrong")
+	}
+
+	prevTier := memberstore.Get().RemoveMember(pmm.Addr)
+	if prevTier == "" {
+		panic("member not found, so cannot be promoted")
+	}
+
+	if prevTier != pmm.FromTier {
+		panic("previous tier changed from the one indicated in the proposal")
+	}
+
+	return memberstore.Get().SetMember(pmm.ToTier, pmm.Addr, pmm.Member)
+}
+
+func NewPromoteMemberRequest(addr std.Address, fromTier string, toTier string) proxy.ProposalRequest {
+	return proxy.ProposalRequest{
+		Title:       "Member Promotion",
+		Description: "This prpoposal is looking to promote a member to an upper tier.",
+		Callback:    promoteMemberCallback,
+		Metadata: &PromoteMemberMetadata{
+			Addr:     addr,
+			FromTier: fromTier,
+			ToTier:   toTier,
+			Member:   memberByTier(toTier),
+		},
+	}
+}
```

Comment:
Should we not check if `toTier > fromTier`?

Reply:
So maybe these functions should be renamed `NewUpdateMemberTierRequest` and `updateMemberTierCallback` (or similar) if they can both promote and demote a member?
Also wdyt about panic-ing if `toTier == fromTier`?